pkg/daemon/ceph/client/filesystem.go

```diff
@@ -96,7 +96,7 @@ func ListFilesystems(context *clusterd.Context, clusterName string) ([]CephFiles
 func GetFilesystem(context *clusterd.Context, clusterName string, fsName string) (*CephFilesystemDetails, error) {
 	args := []string{"fs", "get", fsName}
 	buf, err := NewCephCommand(context, clusterName, args).Run()
-	if err != nil {
+	if err != nil && len(buf) == 0 {
```

Comment:
@travisn I believe that the one and only way that I can think of right now is this. This is because 
`buf, err := NewCephCommand(context, clusterName, args).Run()` returns an `err` and a `byte` array. If there will be no filesystem, the buffer array will have no data so that means the `length` will be 0.

But again, let me know what you think of this.

Reply:
Can we just check on the caller of `GetFilesystem`? If `fs` is nil?

Reply:
So the reason I made the call here is because I wanted to differentiate on the very first call where we are checking if fs exist
That means it's the very first place deep inside under the functions where we're checking for if fs exist or not. So that's why I added it here.
Let me know if that made sense to you

Reply:
If the filesystem doesn't exist, I see a return code of 2 (not found). Let's use that error code. For an example, search Rook for `ENOENT`.
```
# ceph fs get foo
Error ENOENT: filesystem 'foo' not found
[root@rook-ceph-tools-7f96779fb9-b8984 /]# echo $?
2
```